File: apps/api/Api.Domain/Document.cs
using System;
using System.Collections.Generic;

namespace Api.Domain
{
    public enum DocumentStatus
    {
        Draft,
        Completed,
        Published
    }

    public class Document : BaseEntity<Guid>
    {
        public Guid ProjectId { get; set; }
        public Project Project { get; set; } = null!;

        public string Name { get; set; } = null!;
        public string Content { get; set; } = "";
        public DocumentStatus Status { get; set; } = DocumentStatus.Draft;

        public Guid CreatedById { get; set; }
        public Profile CreatedBy { get; set; } = null!;

        public Guid UpdatedById { get; set; }
        public Profile UpdatedBy { get; set; } = null!;

        public ICollection<DocumentCategory> DocumentCategories { get; set; } = new List<DocumentCategory>();
        public ICollection<PublishedDocument> PublishedDocuments { get; set; } = new List<PublishedDocument>();
    }
}

File: apps/web/types/document.ts
export type DocumentStatus = 'Draft' | 'Completed' | 'Published'

export interface Category {
  id: number
  name: string
}

export interface Document {
  id: string
  projectId: string
  projectName: string
  name: string
  content: string
  status: DocumentStatus
  createdAt: string
  updatedAt: string | null
  categories: Category[]
  isPublished: boolean
  currentVersion?: string | null
}

import { Project } from './project'

export interface DocumentIndicator {
  totalDocuments: number
  publishedDocuments: number
  completedDocuments: number
  draftDocuments: number
}

export interface DocumentListResponse {
  documents: Document[]
  indicators: DocumentIndicator
}

export interface DocsNavigationResponse {
  projects: Project[]
  documents: Document[]
}

export interface PublishedDocument {
  id: string
  documentId: string
  version: string
  content: string
  publishedBy: string
  createdAt: string
}

export interface CreateDocumentRequest {
  projectId: string
  name: string
  content: string
  status: string
  categoryIds: number[]
}

export interface UpdateDocumentRequest {
  name: string
  content: string
  status: string
  categoryIds: number[]
}
